src/main/kotlin/goldimax/tesseract/Forward.kt

```diff
@@ -25,43 +27,46 @@ class Forward(private val uniBot: UniBot) {
     private fun handleQQ(): suspend GroupMessage.(String) -> Unit = lambda@{
         if (drive) return@lambda
 
-        val connect = connect.find { x -> x.qq == subject.id }
+        val connect = connect.find { x -> x.qq == subject.id } ?: return@lambda
 
         fun getNick(qq: Member) = when {
             qq.nameCard.isNotEmpty() -> qq.nameCard
             qq.nick.isNotEmpty() -> qq.nick
             else -> qq.id.toString()
         }
 
-        if (connect != null) {
-            val tGroup = connect.tg
-            message.forEach {
-                val msg = StringBuilder()
-                when (it) {
-                    is PlainText -> msg.append(it)
-                    is Image ->
-                        uniBot.tg.sendPhoto(tGroup, it.url(), "${getNick(sender)}: ")
-                    is At -> msg.append(it.display).append(" ")
-                    is AtAll -> msg.append(AtAll.display).append(" ")
-                    is QuoteReply -> msg.append("[Reply👆")
-                        .append(getNick(subject.members[it.source.fromId]))
-                        .append(": ")
-                        .append(it.source.originalMessage.contentToString())
-                        .append("]")
-                    is Face -> msg.append(it.contentToString())
-                    is ForwardMessage ->
-                        msg.append("[Forward]\n").append(it.nodeList.joinToString("\n") {
-                            "${it.senderName}: ${it.message.contentToString()}"
-                        })
-                    is FlashImage -> {
-                        if (forwardFlash) {
-                            uniBot.tg.sendPhoto(tGroup, it.image.url(), "${getNick(sender)}: [闪照]")
-                        } else msg.append("[闪照]")
-                    }
-                    is RichMessage -> msg.append("{").append(it.content).append("}")
+        val tGroup = connect.tg
+        message.forEach { msg ->
+            logger.debug("forward qq $msg")
+
+            val msgStringBuilder = StringBuilder()
+            when (msg) {
+                is PlainText -> msgStringBuilder.append(msg)
+                is Image ->
+                    uniBot.tg.sendPhoto(tGroup, msg.url(), "${getNick(sender)}: ")
+                is At -> msgStringBuilder.append(msg.display).append(" ")
+                is AtAll -> msgStringBuilder.append(AtAll.display).append(" ")
+                is QuoteReply -> msgStringBuilder.append("[Reply👆")
+                    .append(getNick(subject.members[msg.source.fromId]))
+                    .append(": ")
+                    .append(msg.source.originalMessage.contentToString())
+                    .append("]")
+                is Face -> msgStringBuilder.append(msg.contentToString())
+                is ForwardMessage ->
+                    msgStringBuilder.append("[Forward]\n").append(msg.nodeList.joinToString("\n") {
```

Comment:
Use [`joinTo`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/join-to.html) instead of `joinToString`, and pass `msgStringBuilder` as the buffer argument.